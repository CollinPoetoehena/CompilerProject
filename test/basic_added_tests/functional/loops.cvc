extern void printInt(int val);
extern void printNewlines(int val);

export int main() {
    int a = 0;
    int b = 0;

    for (int b = 0, 10, 2) {
        a = a + b;
        printInt(b);
        printNewlines(1);
        printInt(a);
        printNewlines(1);
    }

    do {
        a = a + 7;
        printInt(a);
        printNewlines(1);
    } while (a < 100);

    if (true) {
        do {
            a = a + 7;
            printInt(a);
            printNewlines(1);
        } while (a < 100);
    }

    while (a * a < 20000) {
        while (b < 5) {
            b = b + 1;
            printInt(b);
            printNewlines(1);
        }
        a = a * 2;
        printInt(a);
        printNewlines(1);
    }

    printInt(a);
    printNewlines(1);

    return 0;
}
