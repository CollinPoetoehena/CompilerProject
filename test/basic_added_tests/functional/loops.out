0
0
2
2
4
6
6
12
8
20
27
34
41
48
55
62
69
76
83
90
97
104
111
1
2
3
4
5
222
222